src/Sarif/Schemata/sarif-2.1.0-rtm.5.json

```diff
@@ -1292,6 +1292,7 @@
           "description": "Value that distinguishes this location from all other locations within a single result object.",
           "type": "integer",
           "minimum": -1,
+          "maximum": 2147483647,
```

Comment:
>maximum [](http://example.com/codeflow?start=11&length=7)

nit: add the other fields from SARIF that only accepts non-negative integers besides -1, for example, itemCount, offset, etc.

Reply:
This will prevent us from changing over and over due to a max int value

Reply:
agree, I have created a issue to track
[https://github.com/microsoft/sarif-sdk/issues/2442](https://github.com/microsoft/sarif-sdk/issues/2442)

there can be many to go over, 
after we agree with this proposed way of fix in this PR, I can work on it.

Reply:
You can't just update this schema in place, it has to be versioned, this is an official output of the SARIF technical committee. 

Why have you chosen this max value? Does it correspond to what the SARIF spec allows?

Reply:
Why have you chosen this max value? Does it correspond to what the SARIF spec allows?  
------ the spec just say `non-negative integer`.
[https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317672](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317672)

I don't see in the spec, but do we have definition of the term `integer` or is it always assume to be 32 bit,
I see our C# code object model all use `int` which is 32 bit:
public virtual `int` Id { get; set; }
looks like we should have defined some where,